server/container_create_linux.go

```diff
@@ -921,6 +904,57 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrfactory.Cont
 	return ociContainer, nil
 }
 
+func (s *Server) getImageInfo(userRequestedImage string) (imageResult *storage.ImageResult, err error) {
+	var imgResult *storage.ImageResult
+	if id := s.StorageImageServer().HeuristicallyTryResolvingStringAsIDPrefix(userRequestedImage); id != nil {
+		imgResult, err = s.StorageImageServer().ImageStatusByID(s.config.SystemContext, *id)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		potentialMatches, err := s.StorageImageServer().CandidatesForPotentiallyShortImageName(s.config.SystemContext, userRequestedImage)
+		if err != nil {
+			return nil, err
+		}
+		var imgResultErr error
+		for _, name := range potentialMatches {
+			imgResult, imgResultErr = s.StorageImageServer().ImageStatusByName(s.config.SystemContext, name)
+			if imgResultErr == nil {
+				break
+			}
+		}
+		if imgResultErr != nil {
+			return nil, imgResultErr
+		}
+	}
+	// At this point we know userRequestedImage is not empty; "" is accepted by neither HeuristicallyTryResolvingStringAsIDPrefix
+	// nor CandidatesForPotentiallyShortImageName. Just to be sure:
+	if userRequestedImage == "" {
+		return nil, errors.New("internal error: successfully found an image, but userRequestedImage is empty")
+	}
+
+	return imgResult, nil
+}
+
+func configureTimezone(tz, containerRunDir, mountPoint, mountLabel, etcPath, containerID string, options []string, ctr ctrfactory.Container) error {
+	localTimePath, err := timezone.ConfigureContainerTimeZone(tz, containerRunDir, mountPoint, etcPath, containerID)
+	if err != nil {
+		return fmt.Errorf("setting timezone for container %s: %w", containerID, err)
+	}
+	if localTimePath != "" {
+		if err := securityLabel(localTimePath, mountLabel, false, false); err != nil {
+			return err
+		}
+		ctr.SpecAddMount(rspec.Mount{
+			Destination: "/etc/localtime",
+			Type:        "bind",
+			Source:      localTimePath,
+			Options:     append(options, "bind"),
```

Comment:
Should this be made read-only?

```suggestion
			Options:     []string{"ro", "bind"},
```

Or perhaps even the whole shebang:

```suggestion
			Options:     []string{"ro", "bind", "nodev", "nosuid", "noexec"},
```

Reply:
Not sure what is our take on such system-wide files? Perhaps also for a privileged container, this should be made read/write?

Reply:
Yeah, that's a good point. I don't think we need `r/w`, let me change that to `Options:     append(options, "ro", "bind", "nodev", "nosuid", "noexec"),`

Reply:
I may be wrong here, but I'm pretty sure we're copying the host tz file to the container, or symlinking from one container path to another: https://github.com/cri-o/cri-o/pull/7589/files#diff-d1b079a5798eb933a9cee5821c9b039d99cd82fe192caf20966f5b734f0dd0a0R58-R73
@sohankunkerkar is that correct? or are we mounting the host path into the container?

Reply:
whoops, I missed that part. @haircommander you are right

Reply:
Nice catch! @haircommander :+1: 

I didn't look at the new library, admittedly.

Reply:
Just to follow up, as I don't know where we are at? Are we keeping the bind mount here? If so, then are we making it also read-only?